.github/workflows/build-website.yml

```diff
@@ -1,75 +1,91 @@
+        run: emcmake cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=NO -DCMAKE_C_FLAGS='-Os -flto' -DCMAKE_CXX_FLAGS='-Os -flto'
       - name: C++ build
         run: emmake cmake --build build --target quick-lint-js-vscode quick-lint-js-vscode-licenses
       - name: C++ install
         run: emmake cmake --install build --component vscode --prefix website/public/demo
+      
+      - name: get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
 
       - name: JS configure
-        run: cd website && yarn
+        working-directory: website
+        run: yarn
+
       - name: JS test
-        run: cd website && yarn test
+        working-directory: website
+        run: yarn test
 
       - name: build site
-        run: cd website && yarn build
+        working-directory: website
+        run: yarn build
+
       - name: bundle site
-        run: cd website && zip -r ../website.zip www
+        working-directory: website
+        run: zip -r ../website.zip www
 
-      - name: set up check-links
+      - name: check links
         run: |
-          python3.7 -m venv ENV
-          ENV/bin/pip install bs4 requests
-      - name: check links (dev)
-        run: |
-          cd website/
-          yarn start localhost 9001 &
-          sleep 0.5  # HACK(strager): Wait for server to start.
-          cd ../
-          ./ENV/bin/python ./tools/check-links http://localhost:9001/
-          kill %1
-        shell: bash
-      - name: check links (apache2)
-        run: |
-          sudo DEBIAN_FRONTEND=noninteractive apt-get install apache2 rsync -y
-          sudo rsync --archive --delete website/www/ /var/www/html/
-          sudo sed -i -e 's/AllowOverride\ None/AllowOverride\ FileInfo/' /etc/apache2/apache2.conf
-          sudo a2enmod headers
-          sudo service apache2 start
-          ./ENV/bin/python ./tools/check-links http://localhost/
-        shell: bash
+          rsync --archive --delete website/www/ /var/www/html/
+          service apache2 start
+          python3 ./tools/check-links http://localhost/
 
-      - name: upload build to workflow
+      - uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: apache-error-log
+          path: /var/log/apache2/error.log
+          if-no-files-found: ignore
+ 
+      - name: Upload build to workflow
         uses: actions/upload-artifact@v2
         with:
           if-no-files-found: error
           name: website-${{ github.sha }}
           path: website/www/
-      - name: upload build to long-term storage
+
+      - name: deploy website
```

Comment:
Must fix: This change is wrong. The last step does *not* deploy anything. It copies files to an artifact server. The `website/deploy.sh` script deploys. Undo this change.

Reply:
sure.